src/SAWScript/MGU.hs

```diff
@@ -1256,6 +1257,26 @@ evalTIWithEnv env tenv m =
     (res, _, []) -> Right res
     (_, _, errs) -> Left errs
 
+-- Check a single statement. (This is an external interface.)
+--
+-- The first two arguments are the starting variable and typedef
+-- environments to use.
+--
+-- The third is a current position, and the fourth is the
+-- context/monad type associated with the execution.
+checkStmt :: Map LName Schema -> Map Name Type -> Pos -> Context -> Stmt -> Either [(Pos, String)] Stmt
+checkStmt env tenv pos ctx stmt = do
+  -- Ugh. This should all be cleaned out further. XXX
```

Comment:
In principle that's a fine idea; I think in practice the volume of issues that would generate would be excessive and generate excessive ticket management overhead (there's 12+ of these in here right now, and this is one file of < 1500 lines)... perhaps when I stop I can file one issue mentioning the stuff that's still left.

Reply:
To be clear, I'm not suggesting that you literally open a ticket for every single TODO in the code. Use your best judgment as to which things are well-defined tasks that warrant their own issues.

My goal is to avoid comments that say "should all be cleaned out further" that don't give a clear indication of what is messy about the code or how one might go about cleaning it out. That runs the risk of someone coming back to the code in five years and not having any context for what the comment was originally about.

Reply:
Ideally any comments like this should be sufficient to recover, if not the intended plan for fixing the problem, what the problem was and why it's a problem. This one is not at all clear and I've replaced it.

(also the some of the mess that was here when I wrote that got resolved before commit; it was more clearly a mess at that point)

Please do flag vague notes like this.